### PR TITLE
Java CI

### PR DESCRIPTION
**Pull request by Luigi96**
### Description
Opening a PR with initial CI changes to build and run tests against Java packages. 

Runs:
- [Build Java Semantic Kernel ](https://github.com/joe-braley/semantic-kernel/actions/runs/5008163361)
- [Run Java Integration Tests and Samples](https://github.com/joe-braley/semantic-kernel/actions/runs/5008185823)

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [ ] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: .github/workflows/java-build.yml [hunk 1]
```diff
@@ -1,24 +1,61 @@
...
+          distribution: microsoft
           cache: maven
 
       - name: Build with Maven
-        run: cd java && ./mvnw -B -Pbug-check clean install --file pom.xml
+        run: ./mvnw -B -Pbug-check -DskipTests -X -Pcompile-jdk${{ matrix.java-versions }} clean install --file pom.xml
+        working-directory: java
+
+      - name: Run tests
+        run: ./mvnw -B -Pbug-check -DskipITs -Pcompile-jdk${{ matrix.java-versions }} test --file pom.xml
```
Comment (johnoliver) [inline .github/workflows/java-build.yml]: skipITs has no effect here

File: .github/workflows/java-integration-tests.yml [hunk 2]
```diff
@@ -1,30 +1,68 @@
...
 
     steps:
-      - uses: actions/checkout@v3
-      - name: Set up JDK 17
-        uses: actions/setup-java@v3
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # Need to use JDK 11 to build for JDK 8
+      - name: Set JDK
```
Comment (dsgrieve) [inline .github/workflows/java-integration-tests.yml:34]: Where the steps are the same between integration-tests.yml and java-build.yml, maybe the step should be in its own yml? For instance create a set-sdk.yml and use `- template: set-sdk.yml`

File: java/api-test/pom.xml [hunk 3]
```diff
@@ -42,4 +37,37 @@
...
+                <maven.compiler.target>1.8</maven.compiler.target>
+                <maven.compiler.testSource>1.8</maven.compiler.testSource>
+                <maven.compiler.testTarget>1.8</maven.compiler.testTarget>
+            </properties>
+        </profile>
+        <profile>
+            <id>compile-jdk11</id>
+            <properties>
+                <jdk.version>11</jdk.version>
+                <maven.compiler.source>11</maven.compiler.source>
```
Comment (dsgrieve) [inline java/api-test/pom.xml]: <maven.compiler.source> should always be 1.8. We want to make sure no Java API creeps in that is not compatible with 1.8. The only exception to this is the samples for which <maven.compiler.source> should be 17. 

Also, it seems to me that these profiles should be in java/pom.xml and would not be needed here. But, maybe I'm missing something.

File: samples/java/semantickernel-samples/pom.xml [hunk 4]
```diff
@@ -50,4 +47,28 @@
         </dependency>
     </dependencies>
 
+    <profiles>
```
Comment (dsgrieve) [inline samples/java/semantickernel-samples/pom.xml]: I think you need to revert this change. We want samples to be only 17.

File: java/api-test/pom.xml (hunk 3 above)
Comment (Luigi96) [reply java/api-test/pom.xml]: I set compiler.source to 1.8 and compiler.testSource to 1.8 on all profiles. Does that sounds right or should I delete the profiles and just set the compiler.source to 1.8 as a global property?

File: .github/workflows/java-integration-tests.yml (hunk 2 above)
Comment (Luigi96) [reply .github/workflows/java-integration-tests.yml:34]: There is no `template` property on Github but there is something called `Composite Action` that has a similar behaviour but it will require some investigation. What do you think?

Comment (dsgrieve) [reply .github/workflows/java-integration-tests.yml:34]: Leave for a future.

**Comment by Luigi96** [comment]
All comments have been resolved. Talking with the team we decided that we will wait until our branch gets merge into upstream and we will create a new PR incorporating this changes.
